herokux/data_source_herokux_addons.go

```diff
@@ -0,0 +1,72 @@
+package herokux
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceHerokuxAddons() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceHerokuxAddonsRead,
+		Schema: map[string]*schema.Schema{
+			"app_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"addon_service_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"addon_ids": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"addon_names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceHerokuxAddonsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*Config).API
+
+	var appID string
+	var addonIDs, addonNames []string
+	addonServiceName := d.Get("addon_service_name")
```

Comment:
Continuing from my comment I left on the `api/platform/app.go` file, please make use of the `heroku-go` client here.

For example: (this is incomplete but a start on what I'm looking for)
```suggestion
	config := meta.(*Config)
	client := config.PlatformAPI
	var diags diag.Diagnostics
	
	appID := getAppID(d)
	if v, ok := d.GetOk("addon_service_name"); ok {
		addons, listErr :=  client.AddOnListByApp(context.TODO(), appID, &v5.ListRange{Descending: true})
		if listErr != nil {
			diags = append(diags, diag.Diagnostic{
				Severity: diag.Error,
				Summary:  fmt.Sprintf("Unable to retrieve addons for app %s", appID),
				Detail:   listErr.Error(),
			})
			return diags
		}
	} else {
		
	}
```

Reply:
Fixed, using `heroku-go` client now.